### PR TITLE
SCUMM: Fix GCC warnings

### PR DESCRIPTION
**Pull request by orgads**
* warning: comparison of integer expressions of different signedness
* memset clearing an object of non-trivial type

### CODE REVIEW

File: engines/scumm/actor.h [hunk 1]
```diff
@@ -70,11 +66,15 @@ struct CostumeData {
 	uint32 heCondMaskTable[16];
 
 	void reset() {
+		animCounter = 0;
+		soundCounter = 0;
+		soundPos = 0;
```
Comment (orgads) [inline engines/scumm/actor.h:71]: This is a small behavior change. Looks harmless to me, but someone who knows these classes better than me should approve.

Comment (athrxx) [reply engines/scumm/actor.h:71]: If these 3 vars don't belong there, please move them back to the constructor. No need for experiments here :-)

**Comment by athrxx** [comment]
Looks good to me👍

File: engines/scumm/players/player_mac_indy3.cpp [hunk 2]
```diff
@@ -848,7 +848,6 @@ I3MPlayer::I3MPlayer(ScummEngine *vm, Audio::Mixer *mixer) : _vm(vm), _mixer(mix
 	assert(_musicChannels);
 	for (int i = 0; i < _numMusicChannels; ++i)
 		_musicChannels[i] = new MusicChannel(this);
-	memset(&_pcmSnd, 0, sizeof(_pcmSnd));
```
Comment (AndywinXp) [inline engines/scumm/players/player_mac_indy3.cpp:851]: Is this removal compensated somewhere else?

Comment (athrxx) [reply engines/scumm/players/player_mac_indy3.cpp:851]: It gets cleared in the constructor. I think I had the memset first, and later I added the constructor which makes the memset obsolete (and causes the non-trivial-object warning)

Comment (AndywinXp) [reply engines/scumm/players/player_mac_indy3.cpp:851]: Oh, I see. Thanks!

**Comment by AndywinXp** [comment]
Looks good, thanks!

**Pull request merged**